### PR TITLE
[Gluten-129] Fixes arrow backend building in GHA

### PR DESCRIPTION
**Pull request by zhouyuan**
Fixes arrow backend building. Action cache is reused to speed up arrow build
The velox testing action will be enabled in next PR

This patch also fixes the cpp code format with clang-format-10

Signed-off-by: Yuan Zhou <yuan.zhou@intel.com>

### CODE REVIEW

**Comment by zhouyuan** [comment]
@weiting-chen can you help to take a look?

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -60,40 +60,40 @@ jobs:
...
+#          cd velox
+#          sudo bash scripts/setup-ubuntu.sh
+#          make VELOX_BUILD_MINIMAL=1  MAX_HIGH_MEM_JOBS=8 TREAT_WARNINGS_AS_ERRORS=0 MAX_LINK_JOBS=8
+#      - name: Run unit tests
+#        run: |
+#          cd cpp/
+#          mkdir -p build
+#          cd build
+#          cmake .. -DBUILD_ARROW=0 -DTESTS=1
+#          make
```
Comment (zhouyuan) [inline .github/workflows/unittests.yml:103]: velox compile is very slow and requires lots of memory, disable this job first

**Review by weiting-chen — APPROVED** [review]
It looks good to me.

File: .github/workflows/unittests.yml (hunk 1 above)
Comment (weiting-chen) [reply .github/workflows/unittests.yml:103]: Lets disable it in this PR and enable it after we fix the issue as well as the git branch to be test.

**Comment by zzcclp** [comment]
@weiting-chen Please use 'squash and merge' mode to merge PR.

File: tools/build_arrow.sh [hunk 2]
```diff
@@ -3,75 +3,19 @@
 set -eu
 
 NPROC=$(nproc)
-
-TESTS=OFF
-BUILD_ARROW=OFF
```
Comment (zhztheplayer) [inline tools/build_arrow.sh:8]: Hi @zhouyuan, was the `BUILD_ARROW=ON` option broken before the PR?

Comment (zhouyuan) [reply tools/build_arrow.sh:8]: no, will add back

**Pull request merged**